holopy/scattering/tests/test_lens.py

```diff
@@ -64,10 +67,10 @@ def test_integrate_over_phi_with_quad_points(self):
         wts = LENSMIE._phi_wts
 
         def func(x):
-            return np.cos(np.pi * x)
+            return np.exp(-3 * np.sin(x))
```

Comment:
The old quadrature was optimized for arbitrary functions on an interval, using Gauss-Legendre points and weights. However, these are not optimal for periodic functions, and the actual integrals we are evaluating are periodic with period 2 pi. So I optimized the phi quadrature for a function with period 2 pi, which gives a 3x improvement in accuracy with the same computational cost.

However, the old function in the test, cos (pi * x), is not 2 pi - periodic, and therefore the quadrature does not perform as well on this function as it did before, and the test did not pass. This is a problem with the test and not the code, so I changed the test. I could have changed it to just cos(x), but  that is a more well-behaved function. Since I needed to change the function, I figured that I might as well make the test more challenging as well.

Reply:
thanks for clarifying!